data/scripts/creaturescripts/player_death.lua

```diff
@@ -1,15 +1,33 @@
+local login = CreatureEvent("login")
+function login.onLogin(player)
+	player:registerEvent("prepareDeath")
+	return true
+end
+
+login:register()
+
+local blesses = CreatureEvent("prepareDeath")
+function blesses.onPrepareDeath(player)
+	local function blessMessage()
```

Comment:
Probably here you should pass the player id and check inside the function if the player exists, so we will avoid any crash.